datastream-server/src/main/java/com/linkedin/datastream/server/Coordinator.java

```diff
@@ -1769,7 +1769,7 @@ private void registerMeter(Meter metric) {
     private void registerKeyedMeter(KeyedMeter metric) {
       String metricName = metric.getName();
       _dynamicMetricsManager.registerMetric(MODULE, metric.getKey(), metricName, com.codahale.metrics.Meter.class);
-      _metricInfos.add(new BrooklinMeterInfo(_coordinator.buildMetricName(MODULE, metricName)));
+      _metricInfos.add(new BrooklinMeterInfo(_coordinator.buildMetricName(MODULE, metric.getKey(), metricName)));
```

Comment:
nothing wrong with what you have done. But can you consider the alternative ,as well, used in other classes?

String prefix = CLASS_NAME + MetricsAware.KEY_REGEX; instead of module ? This is how other classes have handled this scenario.

Reply:
Thank you so much for bringing this to my attention.

This question led me to have a conversation with Celia (the feature author) about how metrics work and when regex-based `BrooklinMetricInfo` objects should be used. It turns out there are quite a few interesting details that got me to revisit the way some of the Coordinator metrics are registered/created. I'll put together a short writeup to summarize the key aspects and best practices.

As far as this code change is concerned, here are the adjustments I made:
1. All `KeyedMeter` objects have been changed to be lazily created. This involves:
    - Creating them lazily using `DynamicMetricsManager.createOrUpdate*()`  as opposed to `DynamicMetricsManager.register*()`)
    - Adding regex-based `BrooklinMetricInfo` for them as you suggested
2. Moving the metric below to be a `Meter` instead of a `KeyedMeter` since it doesn't follow the same naming pattern
    ```java
   PERFORM_CLEANUP_ORPHAN_CONNECTOR_TASKS("performCleanupOrphanConnectorTasks", "numOrphanConnectorTasks"),

    ```